### PR TITLE
fix(engine): treat static assignee values as string

### LINKED ISSUE #14109: Static user task assignee value treated as number (leads to an exception)
**Describe the bug**

When the user task assignee is set to a static number value (not a FEEL expression), the activation of the user task leads to an exception.

image

Exception:
```
Expected result of the expression '1547044236679495680' to be 'STRING', but was 'NUMBER'.
```

**To Reproduce**

1. Create a model with an assignee of a user task set to a static number value, e.g. `12345567`.
2. Deploy the model.
3. Start an instance of the model.
4. Let the instance reach the user task.

**Expected behavior**

The user task is executed and the task's assignee is set to `12345567` (string value).

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
08:01:33.484 [] INFO  io.camunda.zeebe.test - Test failed, following records were exported:
08:01:33.610 [] INFO  io.camunda.zeebe.test - Compact log representation:
--------
	['C'ommand/'E'event/'R'ejection] [valueType] [intent] - #[position]->#[source record position] K[key] - [summary of value]
	P9K999 - key; #999 - record position; "ID" element/process id; @"elementid"/[P9K999] - element with ID and key
	Keys are decomposed into partition id and per partition key (e.g. 2251799813685253 -> P1K005). If single partition, the partition is omitted.
	Long IDs are shortened (e.g. 'startEvent_5d56488e-0570-416c-ba2d-36d2a3acea78' -> 'star..acea78'
--------
C DPLY     CREATE         - #01-> -1  -1 - 
E PROC     CREATED        - #02->#01 K01 - process.xml -> "process" (version:1)
E DPLY     CREATED        - #03->#01 K02 - process.xml
C CREA     CREATE         - #04-> -1  -1 - new <process "process"> (default start)  (no vars)
C PI       ACTIVATE       - #05->#04 K03 - PROCESS "process" in <process "process"[K03]>
E CREA     CREATED        - #06->#04 K04 - new <process "process"> (default start)  (no vars)
E PI       ACTIVATING     - #07->#04 K03 - PROCESS "process" in <process "process"[K03]>
E PI       ACTIVATED      - #08->#04 K03 - PROCESS "process" in <process "process"[K03]>
C PI       ACTIVATE       - #09->#04  -1 - START_EVENT "startEv..da0e8bd" in <process "process"[K03]>
E PI       ACTIVATING     - #10->#04 K05 - START_EVENT "startEv..da0e8bd" in <process "process"[K03]>
E PI       ACTIVATED      - #11->#04 K05 - START_EVENT "startEv..da0e8bd" in <process "process"[K03]>
C PI       COMPLETE       - #12->#04 K05 - START_EVENT "startEv..da0e8bd" in <process "process"[K03]>
E PI       COMPLETING     - #13->#04 K05 - START_EVENT "startEv..da0e8bd" in <process "process"[K03]>
E PI       COMPLETED      - #14->#04 K05 - START_EVENT "startEv..da0e8bd" in <process "process"[K03]>
E PI       SEQ_FLOW_TAKEN - #15->#04 K06 - SEQUENCE_FLOW "sequenc..22d1677" in <process "process"[K03]>
C PI       ACTIVATE       - #16->#04 K07 - USER_TASK "task" in <process "process"[K03]>
E PI       ACTIVATING     - #17->#04 K07 - USER_TASK "task" in <process "process"[K03]>
E INCIDENT CREATED        - #18->#04 K08 - EXTRACT_VALUE_ERROR Expected result of the expression '1234567891011121314' to be 'STRING', but was 'NUMBER'.,  @"task"[K07] in <process "process"[K03]>

-------------- Deployed Processes ----------------------
process.xml -> "process" (version:1)[K01] ------
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<definitions xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" xmlns:ns0="http://camunda.org/schema/zeebe/1.0" exporter="Zeebe BPMN Model" exporterVersion="${project.version}" id="definitions_75c21f8c-968a-4322-a009-827562ef7cd8" xmlns:modeler="http://camunda.org/schema/modeler/1.0" modeler:executionPlatform="Camunda Cloud" modeler:executionPlatformVersion="${project.version}" targetNamespace="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns="http://www.omg.org/spec/BPMN/20100524/MODEL">
  <process id="process" isExecutable="true">
    <startEvent id="startEvent_61a36e5e-4c70-4f51-879b-7e626da0e8bd">
      <outgoing>sequenceFlow_acf570d9-e7c2-4b21-bbb0-abfe122d1677</outgoing>
    </startEvent>
    <userTask id="task" name="task">
      <extensionElements>
        <ns0:assignmentDefinition assignee="1234567891011121314"/>
      </extensionElements>
      <incoming>sequenceFlow_acf570d9-e7c2-4b21-bbb0-abfe122d1677</incoming>
      <outgoing>sequenceFlow_af07740c-3c8f-43ba-9dc4-fc17cd763919</outgoing>
    </userTask>
    <sequenceFlow id="sequenceFlow_acf570d9-e7c2-4b21-bbb0-abfe122d1677" sourceRef="startEvent_61a36e5e-4c70-4f51-879b-7e626da0e8bd" targetRef="task"/>
    <endEvent id="endEvent_534843ee-82a6-4517-8f36-227ccc58fb59">
      <incoming>sequenceFlow_af07740c-3c8f-43ba-9dc4-fc17cd763919</incoming>
    </endEvent>
    <sequenceFlow id="sequenceFlow_af07740c-3c8f-43ba-9dc4-fc17cd763919" sourceRef="task" targetRef="endEvent_534843ee-82a6-4517-8f36-227ccc58fb59"/>
  </process>
  <bpmndi:BPMNDiagram id="BPMNDiagram_2efdf830-44de-4ca3-bc61-fe5370e0e2ae">
    <bpmndi:BPMNPlane bpmnElement="process" id="BPMNPlane_d580e40c-3af7-4444-9a9c-b3b39ab750ff">
      <bpmndi:BPMNShape bpmnElement="startEvent_61a36e5e-4c70-4f51-879b-7e626da0e8bd" id="BPMNShape_7f7e5e4a-d4c3-4654-acac-0c086e6008b5">
        <dc:Bounds height="36.0" width="36.0" x="100.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape bpmnElement="task" id="BPMNShape_8854f338-096c-4ef7-aaaf-18f865708d8b">
        <dc:Bounds height="80.0" width="100.0" x="186.0" y="78.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_acf570d9-e7c2-4b21-bbb0-abfe122d1677" id="BPMNEdge_061e7ea7-c454-4b68-a8b9-a68d5189a850">
        <di:waypoint x="136.0" y="118.0"/>
        <di:waypoint x="186.0" y="118.0"/>
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNShape bpmnElement="endEvent_534843ee-82a6-4517-8f36-227ccc58fb59" id="BPMNShape_d3ae1f10-bb2c-492c-9616-61628bdddd14">
        <dc:Bounds height="36.0" width="36.0" x="336.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_af07740c-3c8f-43ba-9dc4-fc17cd763919" id="BPMNEdge_2edec7f1-95a7-4f99-9724-097f787280db">
        <di:waypoint x="286.0" y="118.0"/>
        <di:waypoint x="336.0" y="118.0"/>
      </bpmndi:BPMNEdge>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
</definitions>


--------------- Decomposed keys (for debugging) -----------------
 -1 <-> -1
K01 <-> 2251799813685249
K02 <-> 2251799813685250
K03 <-> 2251799813685251
K04 <-> 2251799813685252
K05 <-> 2251799813685253
K06 <-> 2251799813685254
K07 <-> 2251799813685255
K08 <-> 2251799813685256


io.camunda.zeebe.test.util.stream.StreamWrapperException: No event found matching the criteria

	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at io.camunda.zeebe.test.util.stream.StreamWrapper.getFirst(StreamWrapper.java:91)
	at io.camunda.zeebe.engine.processing.bpmn.activity.UserTaskTest.shouldCreateJobWithAssigneeNumberAsStringHeader(UserTaskTest.java:322)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:232)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:55)
```

</p>
</details>

**Environment:**
- OS:  linux
- Zeebe Version:  8.2.12
- Configuration:  n/a

### PR DESCRIPTION
**Pull request by tmetzke**
## Description

Allow static values for the user task assignee other than String values.

## Related issues

closes #14109 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by tmetzke** [comment]
## Solution options

1. Adjust the static input value in the internal model representation upon transforming the BPMN input.
2. Treat and convert static number expressions for the assignee value as strings wherever they are evaluated in the engine.

## Decision

* I went for option no. 1 since this is consistent with how we treat other user task values (e..g candidate groups and users) that come from the BPMN configuration.
* This option also makes it easier for all code evaluating the assignee expression since it already has the expected target type. Otherwise, all code evaluating the expression would have to know it needs to be treated as a string.

File: engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/UserTaskTransformerTest.java [hunk 1]
```diff
@@ -59,8 +59,9 @@ Stream<Arguments> assignees() {
             Arguments.of(null, null),
             Arguments.of("", null),
             Arguments.of(" ", null),
-            Arguments.of("frodo", "frodo"),
-            Arguments.of("=ring.bearer", "ring.bearer"));
+            Arguments.of("frodo", "\"frodo\""),
+            Arguments.of("=ring.bearer", "ring.bearer"),
+            Arguments.of("12345678", "\"12345678\""));
```
Comment (koevskinikola) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/UserTaskTransformerTest.java:64]: :thought_balloon: Just adding a clarifying comment: the expected change from `Arguments.of("frodo", "frodo")` to `Arguments.of("frodo", "\"frodo\"")` isn't a breaking one. The parsed expression is an intermediate result that is eventually resolved to the same static expression.

**Review by koevskinikola — APPROVED** [review]
Nice one @tmetzke, everything looks great! :tada: 

bors merge

File: engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/UserTaskTransformerTest.java (hunk 1 above)
Comment (tmetzke) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/UserTaskTransformerTest.java:64]: Thanks for making this explicit, @koevskinikola 👍

**Pull request merged**